src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Signature.java

```diff
@@ -809,7 +814,7 @@ private static KnownOIDs getDigestEnum(String algorithm)
         }
     }
 
-    private static byte[] asn1ToDSA(byte[] sig) throws SignatureException {
+    private byte[] asn1ToDSA(byte[] sig) throws SignatureException {
```

Comment:
Have you considered keeping this as a static method but add one more int argument, i.e. signature length? It seems that the method asn1ToECDSA() can be made static too.

Reply:
Ok